### PR TITLE
dm-master: check etcd wal dir when restart

### LINKED ISSUE #1671: dm-master may restarted fail
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do? If possible, provide a recipe for reproducing the error.

    restart one of DM-master members

2. What did you expect to see?

    restarted successfully.

3. What did you see instead?

    restarted failed, with some logs:

    ```
    /dm-master --data-dir=/var/lib/dm-master --name=scale-in-dm-dm-master-2 --peer-urls=http://0.0.0.0:8291 --advertise-peer-urls=http://scale-in-dm-dm-master-2.scale-in-dm-dm-master-peer:8291 --master-addr=:8261 --advertise-addr=scale-in-dm-dm-master-2.scale-in-dm-dm-master-peer:8261 --config=/etc/dm-master/dm-master.toml  --join=http://scale-in-dm-dm-master-4.scale-in-dm-dm-master-peer:8261,http://scale-in-dm-dm-master-2.scale-in-dm-dm-master-peer:8261
    [2021/05/11 10:57:12.898 +08:00] [INFO] [printer.go:54] ["Welcome to dm-master"] ["Release Version"=v2.0.2] ["Git Commit Hash"=c193f6720f16ed86bc97eaa139174dcef3eadc06] ["Git Branch"=heads/refs/tags/v2.0.2] ["UTC Build Time"="2021-04-09 13:32:11"] ["Go Version"="go version go1.13 linux/amd64"]
    [2021/05/11 10:57:12.898 +08:00] [INFO] [main.go:61] ["dm-master config"="{\"log-level\":\"info\",\"log-file\":\"\",\"log-format\":\"text\",\"log-rotate\":\"\",\"rpc-timeout\":\"30s\",\"rpc-rate-limit\":10,\"rpc-rate-burst\":40,\"master-addr\":\":8261\",\"advertise-addr\":\"scale-in-dm-dm-master-2.scale-in-dm-dm-master-peer:8261\",\"config-file\":\"/etc/dm-master/dm-master.toml\",\"name\":\"scale-in-dm-dm-master-2\",\"data-dir\":\"/var/lib/dm-master\",\"peer-urls\":\"http://0.0.0.0:8291\",\"advertise-peer-urls\":\"http://scale-in-dm-dm-master-2.scale-in-dm-dm-master-peer:8291\",\"initial-cluster\":\"scale-in-dm-dm-master-2=http://scale-in-dm-dm-master-2.scale-in-dm-dm-master-peer:8291\",\"initial-cluster-state\":\"new\",\"join\":\"http://scale-in-dm-dm-master-4.scale-in-dm-dm-master-peer:8261,http://scale-in-dm-dm-master-2.scale-in-dm-dm-master-peer:8261\",\"auto-compaction-mode\":\"periodic\",\"auto-compaction-retention\":\"1h\",\"quota-backend-bytes\":2147483648,\"v1-sources-path\":\"\",\"ssl-ca\":\"\",\"ssl-cert\":\"\",\"ssl-key\":\"\",\"cert-allowed-cn\":null}"]
    [2021/05/11 10:57:12.898 +08:00] [INFO] [config.go:440] ["Set log level of etcd to `warn`, if you want to log more message about etcd, change log-level to `debug` in master configuration file"]
    [2021/05/11 10:57:12.898 +08:00] [INFO] [server.go:150] ["config after join prepared"] [config="{\"log-level\":\"info\",\"log-file\":\"\",\"log-format\":\"text\",\"log-rotate\":\"\",\"rpc-timeout\":\"30s\",\"rpc-rate-limit\":10,\"rpc-rate-burst\":40,\"master-addr\":\":8261\",\"advertise-addr\":\"scale-in-dm-dm-master-2.scale-in-dm-dm-master-peer:8261\",\"config-file\":\"/etc/dm-master/dm-master.toml\",\"name\":\"scale-in-dm-dm-master-2\",\"data-dir\":\"/var/lib/dm-master\",\"peer-urls\":\"http://0.0.0.0:8291\",\"advertise-peer-urls\":\"http://scale-in-dm-dm-master-2.scale-in-dm-dm-master-peer:8291\",\"initial-cluster\":\"\",\"initial-cluster-state\":\"existing\",\"join\":\"http://scale-in-dm-dm-master-4.scale-in-dm-dm-master-peer:8261,http://scale-in-dm-dm-master-2.scale-in-dm-dm-master-peer:8261\",\"auto-compaction-mode\":\"periodic\",\"auto-compaction-retention\":\"1h\",\"quota-backend-bytes\":2147483648,\"v1-sources-path\":\"\",\"ssl-ca\":\"\",\"ssl-cert\":\"\",\"ssl-key\":\"\",\"cert-allowed-cn\":null}"]
    2021-05-11 10:57:12.899541 W | pkg/fileutil: check file permission: directory "/var/lib/dm-master" exist, but the permission is "drwxrwxrwx". The recommended permission is "-rwx------" to prevent possible unprivileged access to the data.
    [2021/05/11 10:57:12.925 +08:00] [ERROR] [main.go:69] ["fail to start dm-master"] [error="[code=38038:class=dm-master:scope=internal:level=high], Message: fail to start embed etcd, RawCause: couldn't find local name \"scale-in-dm-dm-master-2\" in the initial cluster configuration"] [errorVerbose="[code=38038:class=dm-master:scope=internal:level=high], Message: fail to start embed etcd, RawCause: couldn't find local name \"scale-in-dm-dm-master-2\" in the initial cluster configuration\ngithub.com/pingcap/dm/pkg/terror.(*Error).Delegate\n\t/home/jenkins/agent/workspace/build_dm_multi_branch_v2.0.2/go/src/github.com/pingcap/dm/pkg/terror/terror.go:309\ngithub.com/pingcap/dm/dm/master.startEtcd\n\t/home/jenkins/agent/workspace/build_dm_multi_branch_v2.0.2/go/src/github.com/pingcap/dm/dm/master/etcd.go:56\ngithub.com/pingcap/dm/dm/master.(*Server).Start\n\t/home/jenkins/agent/workspace/build_dm_multi_branch_v2.0.2/go/src/github.com/pingcap/dm/dm/master/server.go:199\nmain.main\n\t/home/jenkins/agent/workspace/build_dm_multi_branch_v2.0.2/go/src/github.com/pingcap/dm/cmd/dm-master/main.go:67\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:203\nruntime.goexit\n\t/usr/local/go/src/runtime/asm_amd64.s:1357"]
    ```

4. Versions of the cluster

    - DM version (run `dmctl -V` or `dm-worker -V` or `dm-master -V`):

        ```
        v2.0.2
        ```

---

some highlights:

- config for the binary
    ```
    --join=http://scale-in-dm-dm-master-4.scale-in-dm-dm-master-peer:8261,http://scale-in-dm-dm-master-2.scale-in-dm-dm-master-peer:8261
    ```
- config after adjusted by dm-master
    ```
    "initial-cluster":"scale-in-dm-dm-master-2=http://scale-in-dm-dm-master-2.scale-in-dm-dm-master-peer:8291",
    "initial-cluster-state":"new",
    "join":"http://scale-in-dm-dm-master-4.scale-in-dm-dm-master-peer:8261,http://scale-in-dm-dm-master-2.scale-in-dm-dm-master-peer:8261",
    ```
- config after prepared with join file and directory
    ```
    "initial-cluster":"",
    "initial-cluster-state":"existing",
    "join":"http://scale-in-dm-dm-master-4.scale-in-dm-dm-master-peer:8261,http://scale-in-dm-dm-master-2.scale-in-dm-dm-master-peer:8261",
    ```

according to the prepared config, we can assume the following code path was triggered: https://github.com/pingcap/dm/blob/c193f6720f16ed86bc97eaa139174dcef3eadc06/dm/master/etcd.go#L108-L113

- but from the error reported by the etcd, the following call stack may be triggered:
      - https://github.com/etcd-io/etcd/blob/ae9734ed278b7a1a7dfc82e800471ebbf9fce56f/etcdserver/config.go#L207
      - https://github.com/etcd-io/etcd/blob/ae9734ed278b7a1a7dfc82e800471ebbf9fce56f/etcdserver/config.go#L192
      - https://github.com/etcd-io/etcd/blob/ae9734ed278b7a1a7dfc82e800471ebbf9fce56f/etcdserver/server.go#L349
- then it means `haveWAL := wal.Exist(cfg.WALDir())` is `false`, or in other words, `cfg.WALDir()` (`{DataDir}/member/wal`) is not exist.
- but in DM, we only check `{DataDir}/member` is exist https://github.com/pingcap/dm/blob/c193f6720f16ed86bc97eaa139174dcef3eadc06/dm/master/etcd.go#L109, so it's caused when restarting with `{DataDir}/member` but without `{DataDir}/member/wal` and we may need to check `{DataDir}/member/wal` instead?

### PR DESCRIPTION
**Pull request by GMHDBJD**
### What problem does this PR solve? 
close #1671 

### What is changed and how it works?
check wal dir exist

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Review by lance6716 — APPROVED** [review]

**Comment by GMHDBJD** [comment]
need fix ut and add new logic, WIP

File: .github/workflows/check-and-build.yml [hunk 1]
```diff
@@ -82,5 +82,5 @@ jobs:
       - name: GolangCI Lint
         uses: golangci/golangci-lint-action@v2
         with:
-          version: latest
+          version: v1.39
```
Comment (GMHDBJD) [inline .github/workflows/check-and-build.yml:85]: Use latest after we upgrade to go1.16, otherwise local build with go1.13 will fail

**Comment by GMHDBJD** [comment]
@csuzhangxc PTAL. Don't know whether fix the issue.

**Review by csuzhangxc — APPROVED** [review]
LGTM

**Pull request merged**